src/compiler/core.ts

```diff
@@ -1710,6 +1710,10 @@ namespace ts {
         return moduleResolution;
     }
 
+    export function getStrictOptionValue(compilerOptions: CompilerOptions, flag: "noImplicitAny" | "noImplicitThis" | "strictNullChecks" | "strictFunctionTypes" | "alwaysStrict"): boolean {

```

Comment:
Is there some way we can brand `CompilerOptions` internally in the type system such that we cannot use the `strict`-depending fields directly - it must be used through this function?

Reply:
Could not think of a way to do that without changing the shape of the compiler options object..

Reply:
Documenting exploration for future reference: You can move members into a superclass/superinterface, then override them (marked `@internal`) with an intersection with `never` to change the field type to `never` internally (yet this is not an error for very odd reasons) - however in practice this messes with unit tests (some of which expect to explicitly set fields of those types) and doesn't prevent you from doing things like saying `if (compilerOptions.noImplicitAny)`, which was the goal (as a `never` member is still considered present) :(